### PR TITLE
chore(platform): add indexing file for `carbon-web-components`

### PR DESCRIPTION
**Pull request by kennylam**
### Description

Index the `carbon-web-components` package with Carbon Platform.

@francinelucca For `externalDocsUrl` and `usage`/`style`/`a11y` paths, I've kept the links pointing to Carbon React documentation. Since `carbon-web-components` should (eventually) mirror the React components, I think it's a good idea to treat those as the canonical docs? Let me know what you think. Also please let me know if there is anything missing, for instance icons. Not sure where those should be added and if we even have one we should be using.

### Changelog

**New**

- `carbon.yml` index file for `carbon-web-components` package

### CODE REVIEW

**Comment by francinelucca** [comment]
@kennylam pointing to `carbondesignsystem.com`'s documentation sounds good to me!

P.S.: since you're inheriting `carbon-styles` and `docs` is an inheritable property (see [platform's schema](https://next.carbondesignsystem.com/contributing/schema#asset-keys)) , you can forego writing the docs explicitly here and it will be passed down automatically as long as the asset ids match what's defined in https://github.com/carbon-design-system/carbon/blob/main/packages/styles/carbon.yml (this is not true for `externalDocsUrl` since that is not an inheritable property) 

Also if the components look more or less the same feel free to borrow some thumbs from [core](https://github.com/carbon-design-system/carbon/tree/main/packages/styles/thumbnails) and add the corresponding `thumbnailPath`

File: packages/carbon-web-components/carbon.yml [hunk 1]
```diff
@@ -0,0 +1,510 @@
+# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas@v1/carbon-resources.schema.json
+library:
+  id: carbon-web-components
+  name: Carbon Web Components
+  description:
+  description: Build user interfaces with web components based on Carbon's React library.
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:6]:
```suggestion
  description: Build user interfaces with web components based on Carbon's React library.
```

File: packages/carbon-web-components/carbon.yml [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+  combo-box:
+    status: stable
+    framework: web-component
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-combo-box
+    docs:
+  content-switcher:
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:89]:
```suggestion
  combo-box:
    status: stable
    framework: web-component
    demoLinks:
      - type: storybook
        name: Storybook
        action: link
        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-combo-box
  content-switcher:
```

File: packages/carbon-web-components/carbon.yml [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+  copy-button:
+    status: stable
+    framework: web-component
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-copy-button
+    docs:
+  data-table:
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:111]:
```suggestion
  copy-button:
    status: stable
    framework: web-component
    demoLinks:
      - type: storybook
        name: Storybook
        action: link
        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-copy-button
  data-table:
```

File: packages/carbon-web-components/carbon.yml [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-icon
+    docs:
+    tags:
+      - input-control
+      - media
+  inline-loading:
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:177]:
```suggestion
  icon:
    name: Icon
    framework: web-component
    status: stable
    type: component
    demoLinks:
      - type: storybook
        name: Storybook
        action: link
        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-icon
    tags:
      - input-control
      - media
  inline-loading:
```

File: packages/carbon-web-components/carbon.yml [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+  multiselect:
+    status: stable
+    externalDocsUrl: https://www.carbondesignsystem.com/components/dropdown/usage/#multiselect
+    framework: web-component
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-multi-select
+    docs:
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:264]:
```suggestion
  multiselect:
    status: stable
    externalDocsUrl: https://www.carbondesignsystem.com/components/dropdown/usage/#multiselect
    framework: web-component
    demoLinks:
      - type: storybook
        name: Storybook
        action: link
        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-multi-select
```

File: packages/carbon-web-components/carbon.yml [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+  notifications:
+    status: stable
+    externalDocsUrl: https://www.carbondesignsystem.com/components/notification/usage
+    framework: web-component
+    demoLinks:
+      - type: storybook
+        name: Storybook
+        action: link
+        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-notifications
+    docs:
```
Comment (francinelucca) [inline packages/carbon-web-components/carbon.yml:274]:
```suggestion
  notifications:
    status: stable
    externalDocsUrl: https://www.carbondesignsystem.com/components/notification/usage
    framework: web-component
    demoLinks:
      - type: storybook
        name: Storybook
        action: link
        url: https://www.ibm.com/standards/carbon/carbon-web-components/?path=/docs/components-notifications
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Looking good! 🚀, some minor suggestions

**Comment by kennylam** [comment]
@francinelucca got it, thanks!

**Review by francinelucca — APPROVED** [review]
🚀

**Review by ariellalgilmore — APPROVED** [review]

**Pull request merged**